### PR TITLE
Document scale by cluster role

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: docs-source/docs/modules/deployment/pages/operator-reference.adoc [hunk 1]
```diff
@@ -39,6 +40,144 @@ Another way of changing number of Pods is to use `kubectl scale` without changin
 kubectl scale --replicas=3 akkamicroservices shopping-cart-servics
 ----
 
+=== Scale Akka Cluster roles
```
Comment (ignasi35) [inline docs-source/docs/modules/deployment/pages/operator-reference.adoc:43]: Maybe add a note stating the operator doesn't yet support adding multiple roles to a node?

**Review by ignasi35 — APPROVED** [review]

**Comment by patriknw** [comment]
This is ready for reviews. I changed it to draft because we should not merge and publish this yet. We need to change the publishing to only publish when tagged.

File: docs-source/docs/modules/deployment/pages/operator-reference.adoc [hunk 2]
```diff
@@ -39,6 +40,155 @@ Another way of changing number of Pods is to use `kubectl scale` without changin
...
+    replicasRatio: 60
+----
+
+<1> total of 10 replicas means that 10 Pods, corresponding to 10 Akka Cluster members, will be started
+<2> 40% of the Pods will be used for role `entity`
+<3> 60% of the Pods will be used for role `projection`
+
+The `replicasRatio` for a role defines the proportion of the total number of `replicas`, relative to other roles, that will be used for the role. For example, role `entity` with `replicasRatio: 40` and role `projection` with `replicasRatio: 60` means that 10 `replicas` will result in 4 replicas for role `entity` and 6 replicas for role `projection`. Scaling to 15 will result in 6 for `entity` and 9 for `projection`.
+
+In this example we have used the ratio 40:60 because the sum is 100 and therefore corresponds to percentage of the total replicas. However, it's not required to have a sum of 100. Same ratio could be defined as 2:3.
```
Comment (chbatey) [inline docs-source/docs/modules/deployment/pages/operator-reference.adoc:72]: /Same ration/The same ratio/ reads a bit better

File: docs-source/docs/modules/deployment/pages/operator-reference.adoc [hunk 3]
```diff
@@ -39,6 +40,155 @@ Another way of changing number of Pods is to use `kubectl scale` without changin
...
+
+|20
+|10
+|10 (max)
+
+|===
+
+When assigning the number of replicas for each role it is performed in the order they are defined. That is the reason the `entity` role is given 2 out of 2 replicas the `projection` role still has 0. The `entity` role has a minimum preference of 2. In this way you can give priority to certain roles.
+
+If `replicasMax` is defined on all roles it can't be satisfied if the total number of replicas exceeds the sum of `replicasMax` of all roles. For example, if `replicasMax: 10` would have been defined on both `entity` and `projection` and scaling to 25 replicas. In that case the exceeding 5 would be evenly distributed over `entity` and `projection` and result in 13 for `entity` and 12 for `projection`.
```
Comment (chbatey) [inline docs-source/docs/modules/deployment/pages/operator-reference.adoc:190]: Perhaps less useful but more initiative definition of max would be that the extra would have no roles

File: docs-source/docs/modules/how-to/pages/scale-independently.adoc [hunk 4]
```diff
@@ -1,10 +1,8 @@
 = How to scale command-side and query-side independently
-
+:page-supergroup-java-scala: Language
 
 include::partial$include.adoc[]
 
-
-
 Not all nodes in an Akka Cluster need to be identical. When starting each process you can add some metadata to it to give that instance a special purpose. {akka}/typed/cluster.html#node-roles[Node Roles {tab-icon}, window="tab"] in Akka Cluster will handle that metadata.
```
Comment (chbatey) [inline docs-source/docs/modules/how-to/pages/scale-independently.adoc:6]: Maybe point out here that the code/image is the same. Just slightly different behaviour per role

**Review by chbatey — APPROVED** [review]
LGTM

File: docs-source/docs/modules/deployment/pages/operator-reference.adoc [hunk 5]
```diff
@@ -39,6 +40,155 @@ Another way of changing number of Pods is to use `kubectl scale` without changin
...
+
+<1> total of 10 replicas means that 10 Pods, corresponding to 10 Akka Cluster members, will be started
+<2> 40% of the Pods will be used for role `entity`
+<3> 60% of the Pods will be used for role `projection`
+
+The `replicasRatio` for a role defines the proportion of the total number of `replicas`, relative to other roles, that will be used for the role. For example, role `entity` with `replicasRatio: 40` and role `projection` with `replicasRatio: 60` means that 10 `replicas` will result in 4 replicas for role `entity` and 6 replicas for role `projection`. Scaling to 15 will result in 6 for `entity` and 9 for `projection`.
+
+In this example we have used the ratio 40:60 because the sum is 100 and therefore corresponds to percentage of the total replicas. However, it's not required to have a sum of 100. Same ratio could be defined as 2:3.
+
+The Akka Operator will automatically provide the `akka.cluster.roles` configuration to the application.
```
Comment (johanandren) [inline docs-source/docs/modules/deployment/pages/operator-reference.adoc:74]: Maybe also good to mention that the user should not define any roles in their `application.conf` here as well?

File: docs-source/docs/modules/deployment/pages/operator-reference.adoc [hunk 6]
```diff
@@ -39,6 +40,155 @@ Another way of changing number of Pods is to use `kubectl scale` without changin
...
+See xref:how-to:scale-independently.adoc[] for information about what is needed in the application to make use of the Akka Cluster roles.
+
+The Akka Operator adds a label `akka.lightbend.com/cluster-role` to the Pods, which for example can be used to select Pods with a certain role:
+
+[source,shell script]
+----
+kubectl get pods -label akka.lightbend.com/cluster-role=entity
+----
+
+Only one role per Akka Cluster member is currently supported.
```
Comment (johanandren) [inline docs-source/docs/modules/deployment/pages/operator-reference.adoc:85]: Good making that clear

**Review by johanandren — APPROVED** [review]

File: docs-source/docs/modules/deployment/pages/operator-reference.adoc (hunk 3 above)
Comment (patriknw) [reply docs-source/docs/modules/deployment/pages/operator-reference.adoc:190]: interesting thought, but if we would support no-roles I think it should be more explicit, like a special role name. That would actually work already with any role name that isn't used by the application.

**Pull request merged**